DxDispatch/README.md

```diff
@@ -0,0 +1,68 @@
+
+Some of the things you can do with this tool:
+- Launch [DirectML](https://github.com/Microsoft/directml) operators to understand how they work with different inputs.
+- Run custom HLSL compute shaders that are compiled at runtime using the [DirectX Shader Compiler](https://github.com/Microsoft/DirectXShaderCompiler).
+- Debug binding and API usage issues. DxDispatch hooks into the Direct3D and DirectML debug layers and prints errors and warnings directly to the console; no need to attach a debugger.
+- Experiment with performance by benchmarking dispatches.
+- Take GPU or timing captures using [PIX on Windows](https://devblogs.microsoft.com/pix/download/). Labeled events and resources make it easy to correlate model objects to D3D objects.
+
+This tool is *not* designed to be a general-purpose framework for building large computational models or running in production scenarios. The focus is on experimentation and learning!
+
+# DirectX Component Versions
+
+DxDispatch currently targets the following versions of DirectX components:
+
+| Component               | Version                                                                                                |
+| ----------------------- | ------------------------------------------------------------------------------------------------------ |
+| DirectML                | [1.8.0](https://www.nuget.org/packages/Microsoft.AI.DirectML/1.8.0)                                    |
+| Direct3D 12             | [1.4.10 (Agility SDK)](https://www.nuget.org/packages/Microsoft.Direct3D.D3D12/1.4.10)                 |
+| DirectX Shader Compiler | [December 2021 (v1.6.2106)](https://github.com/microsoft/DirectXShaderCompiler/releases/tag/v1.6.2112) |
+
+# System Requirements
+
+DxDispatch is currently only available for Windows 10/11, but support for WSL may be added in the future. DxDispatch depends on the [DirectX 12 Agility SDK](https://devblogs.microsoft.com/directx/announcing-dx12agility/), which is only available on Version 1909+.
+
+- A DirectX 12 capable hardware device.
+- Windows 10 November 2019 Update (Version 1909; Build 18363) or newer.
+- If testing shaders that use Shader Model 6.6:
+  - AMD: [Adrenalin 21.4.1 preview driver](https://www.amd.com/en/support/kb/release-notes/rn-rad-win-21-4-1-dx12-agility-sdk)
+  - NVIDIA: drivers with version 466.11 or higher
+
+Each Windows release of DxDispatch bundles a recent redistributable copy of DirectML (plus debug layers), Direct3D 12 (plus debug layers), and the DX Shader Compiler. You do not need to install any additional software to run the dxdispatch executable, but a recent driver for your DX12 device is recommended.
```

Comment:
Love it. Sure, makes the package bigger, but self-contained is great for sharing repros.